api/src/main/java/org/apache/iceberg/Table.java

```diff
@@ -173,6 +190,13 @@ default String name() {
    */
   UpdateLocation updateLocation();
 
+  /**
+   * Create a new {@link UpdateSnapshotRefs} to update snapshot references and commit the changes.
+   *
+   * @return a new {@link UpdateSnapshotRefs}
+   */
+  UpdateSnapshotRefs updateRefs();
```

Comment:
How do you think the other table mutations should handle refs? For example, an append might look like this:

```java
table.newAppend()
    .appendFile(someFile)
    .commit();
```

Should we add an optional `branch` to all of the `SnapshotProducer` operations?

```java
table.newAppend()
    .appendFile(someFile)
    .branch("some_branch")
    .commit();
```

Curious what you've been thinking.

Reply:
Yes I do have exactly the same thought, where append will go to the `main` branch by default and an override is allowed. I think a follow-up PR is needed for this feature.